thewalrus/reference.py

```diff
@@ -278,3 +282,121 @@ def hafnian(M, loop=False):
         tot_sum = tot_sum + result
 
     return tot_sum
+
+
+def mapper(x, objects):
+    """Helper function to turn a permutation and bistring into an element of rpmp.
+
+    Args:
+        x (tuple): tuple containing a permutation and a bistring.
+        objects (list): list objects to permute
+
+    Returns:
+        tuple: permuted objects
+    """
+    (perm, bit) = x
```

Comment:
will `x` always be a tuple of size 2? Consider adding a check here.

Reply:
Yup, it will always be. Each element of RPMP can be labelled by a permutation and a bitstring, `perm` and `bit`.